src/main/java/org/team1540/robot2018/Robot.java

```diff
@@ -376,4 +204,199 @@ public void teleopPeriodic() {
+    LEFT_DOUBLE_SCALE_THEN_SWITCH(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for the left scale"),
+    LEFT_SCALE_STRAIGHT(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for left scale"),
+    MIDDLE_TO_LEFT_SWITCH(new SingleCubeSwitchAuto("middle_to_left_switch"), "Going for the left switch"),
+    MIDDLE_TO_RIGHT_SWITCH(new SingleCubeSwitchAuto("middle_to_right_switch"), "Going for the right switch"),
+    SWITCH_DOUBLE_CUBE_LEFT(new SwitchDoubleCube("left"), "Going for the left switch"),
+    SWITCH_DOUBLE_CUBE_RIGHT(new SwitchDoubleCube("right"), "Going for the right switch"),
+    RIGHT_HOOK(new SingleCubeSwitchAuto("right_hook"), "Going for the right switch"),
+    RIGHT_SCALE(new ProfileScaleAuto("right_scale"), "Going for the right scale"),
+    DRIVE_TIMED(new DriveTimed(ControlMode.PercentOutput, Tuning.stupidDriveTime, Tuning.stupidDrivePercent), "Driving straight for time");
+
+    private final Command autoCommand;
+    private final String defaultMessage;
+
+    MotionProfile(Command autoCommand, String defaultMessage) {
+      this.autoCommand = autoCommand;
+      this.defaultMessage = defaultMessage;
+    }
+
+  }
+
+  private static class DecisionNode {
+
+    @NotNull
+    final BooleanSupplier condition;
+    @Nullable
+    private Message message = null;
+    @Nullable
+    final MotionProfile profile;
+    @Nullable
+    final DecisionNode[] children;
+
+    public DecisionNode(Posession posession, @NotNull MotionProfile profile) {
+      this.condition = posession.condition;
+      this.profile = profile;
+      this.children = null;
+      message = new Message(profile.defaultMessage);
+    }
+
+    public DecisionNode(@NotNull MotionProfile profile) {
+      this.condition = () -> (true);
+      this.children = null;
+      this.profile = profile;
+    }
+
+    public DecisionNode(Posession posession, @NotNull DecisionNode[] children) {
+      this.condition = posession.condition;
+      this.children = children;
+      this.profile = null;
+    }
+
+    public DecisionNode(@NotNull DecisionNode[] children) {
+      this.condition = () -> (true);
+      this.children = children;
+      this.profile = null;
+    }
+
+    public DecisionNode setMessage(Message message) {
+      this.message = message;
+      return this;
+    }
+
+  }
+
+  private static class Message {
+    private String message;
+    private boolean isError = false;
+
+    public Message(String message) {
+      this.message = message;
+    }
+
+    public Message(String message, boolean isError) {
+      this.message = message;
+      this.isError = isError;
+    }
+
+    public void displayMessage() {
+      if (isError) {
+        DriverStation.reportError(message, false);
```

Comment:
`DriverStation.reportError` when not printing a stack trace is exactly the same thing as writing to `System.err` so I think you could probably do this with a conditional statement

Reply:
I think an if is the cleanest way of doing this still